src/platform/linux/x11/mod.rs

```diff
@@ -954,10 +954,28 @@ impl Window {
         let im = unsafe {
             let _lock = GLOBAL_XOPENIM_LOCK.lock().unwrap();
 
-            let im = (x_events_loop.display.xlib.XOpenIM)(x_events_loop.display.display, ptr::null_mut(), ptr::null_mut(), ptr::null_mut());
+            let open_im = || (x_events_loop.display.xlib.XOpenIM)(
+                x_events_loop.display.display,
+                ptr::null_mut(),
+                ptr::null_mut(),
+                ptr::null_mut(),
+            );
+
+            let mut im = open_im();
```

Comment:
After doing some research, I believe this call should be preceded by an empty `XSetLocaleModifiers`, [as is done in alacritty](https://github.com/jwilm/alacritty/blob/6debc4f3351446417d0c4e38173cd9ef0faa71d5/src/window.rs#L335). Otherwise, the user's `XMODIFIERS` environment variable is never used. This is also done in [GLFW](https://github.com/glfw/glfw/blob/6e69f63ffa4059d0f16344e6b038285c1ca78c16/src/x11_init.c#L944), [GTK](https://github.com/GNOME/gtk/blob/695d141f32d8aed14b57cc065554974d541f8ba6/gtk/gtkimcontextxim.c#L364), and so on.